### PR TITLE
U4-9591 All PetaPoco Relators for 1:M mappings need to be reviewed to…

### PR DESCRIPTION
**Pull request by clausjensen**
… ensure there is a correct sort being used

### CODE REVIEW

**Comment by clausjensen** [comment]
note theres a whole bunch of unit tests failing but i havent looked into it until i get some feedback on what we actually need to do from @Shazwazza

File: src/Umbraco.Core/Persistence/Repositories/ContentTypeBaseRepository.cs [hunk 1]
```diff
@@ -101,7 +101,7 @@ protected IEnumerable<int> PerformGetByQuery(IQuery<PropertyType> query)
 
             var translator = new SqlTranslator<PropertyType>(sqlClause, query);
             var sql = translator.Translate()
-                                .OrderBy<PropertyTypeDto>(x => x.PropertyTypeGroupId, SqlSyntax);
```
Comment (Shazwazza) [inline src/Umbraco.Core/Persistence/Repositories/ContentTypeBaseRepository.cs]: @clausjensen I don't quite think you know what you are doing here ;) This task is to review the relators like GroupPropertyTypeRelator to see what they are comparing on. A relator will take multiple rows and turn it into a 1:M object. For example, the GroupPropertyTypeRelator is comparing on the cmsPropertyTypeGroup.pk value, which means that this result would need to be sorted by that value in order for it to work. I'll fix it up but just want to let you know how this works.

File: src/Umbraco.Core/Persistence/Repositories/DictionaryRepository.cs [hunk 2]
```diff
@@ -96,7 +96,8 @@ protected override Sql GetBaseQuery(bool isCount)
                 sql.Select("*")
                    .From<DictionaryDto>(SqlSyntax)
                    .LeftJoin<LanguageTextDto>(SqlSyntax)
-                   .On<DictionaryDto, LanguageTextDto>(SqlSyntax, left => left.UniqueId, right => right.UniqueId);
+                   .On<DictionaryDto, LanguageTextDto>(SqlSyntax, left => left.UniqueId, right => right.UniqueId)
+                   .OrderBy<DictionaryDto>(x => x.UniqueId, SqlSyntax);
```
Comment (Shazwazza) [inline src/Umbraco.Core/Persistence/Repositories/DictionaryRepository.cs]: Yup! This one is correct

File: src/Umbraco.Core/Persistence/Repositories/ContentTypeBaseRepository.cs (hunk 1 above)
Comment (clausjensen) [inline src/Umbraco.Core/Persistence/Repositories/ContentTypeBaseRepository.cs]: Well as I pointed out in the issue - I had no idea what this task is about or why it was needed as the issue didn't have any context information at all .. it makes a bit more sense now what actually needs to be done :)

**Pull request merged**